Rename xref-pop-marker-stack to xref-go-back to follow Emacs 29.1 changes

sunlin7 opened the pull request:
The Emacs-29.1 had renamed the `xref-pop-marker-stack` to `xref-go-back`,
https://github.com/emacs-mirror/emacs/commit/d8caa3d9fbd90de41efacfeb23c242df81c62bd0

And there will has a warning message when we call the `xref-pop-marker-stack` on emacs-29.1+,

sunlin7 committed:
Rename xref-pop-marker-stack to xref-go-back to follow Emacs 29.1 changes

Co-Author: bcc32

bcc32 commented:
Made some fixes, can you please test that this works for you?

sunlin7 commented:
Hi @bcc32  I had verified  the changes, works on my side. Thank you !

bcc32 commented on layers/+vim/evil-better-jumper/packages.el:
```diff
@@ -30,10 +30,10 @@
     :config
     (better-jumper-mode 1)
-    (spacemacs|hide-lighter better-jumper-mode)
+    (global-set-key [remap xref-go-back] #'better-jumper-jump-backward)
```
I assume this change is unintentional?  (If you could also please rebase your PR when you get the chance, to remove the extraneous bookmark/evil-collection change)

sunlin7 replied:
No, it's not needed. Removed. Thank you!

sunlin7 commented:
@bcc32  Thank you for the comments, and apologize that I rebased with the latest branch and then force push changes had override your additional commit in this PR.  It seems no way to restore that, so I added you to the co-author in the commit message.

bcc32 commented:
No problem at all, thanks for double-checking.
